### PR TITLE
refactor: move queryID generator out of physical plan

### PR DESCRIPTION
**Pull request by agavra**
### Description 
This is a minor refactor which moves queryID generation outside of the physical plan into a place that is better suited for "replace" style queries which reuse the existing queryID. The query id generation is now done in `EngineExecutor` (there is also no queryId generation in `QueryMetadata` anymore.

### Testing done 
Non-functional change, moved unit tests as appropriate. 

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/test/java/io/confluent/ksql/testutils/AnalysisTestUtil.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import io.confluent.ksql.parser.tree.Statement;
 import io.confluent.ksql.planner.LogicalPlanner;
 import io.confluent.ksql.planner.plan.OutputNode;
+import io.confluent.ksql.query.id.SequentialQueryIdGenerator;
```
Comment (vcrfxia) [inline ksqldb-engine/src/test/java/io/confluent/ksql/testutils/AnalysisTestUtil.java:31]: nit: unused import?

File: ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java [hunk 2]
```diff
@@ -83,15 +80,8 @@ public SourceName getIntoSourceName() {
   }
 
   @Override
-  public QueryId getQueryId(final QueryIdGenerator queryIdGenerator) {
-    final String base = queryIdGenerator.getNext().toUpperCase();
```
Comment (vcrfxia) [inline ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java:87]: What happened to this `toUpperCase()` call? I don't see it in the migrated code.

File: ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    final KsqlStructuredDataOutputNode structured = (KsqlStructuredDataOutputNode) outputNode;
+    if (!structured.createInto()) {
+      return new QueryId("INSERTQUERY_" + idGenerator.getNext());
+    }
+
+    final SourceName sink = outputNode.getSinkName().get();
+    final Set<String> queriesForSink = metaStore.getQueriesWithSink(sink);
+    if (queriesForSink.size() > 1) {
+      throw new KsqlException("REPLACE for sink " + sink + " is not supported because there are "
```
Comment (vcrfxia) [inline ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java:64]: Why is there no check here on whether the sink specifies CREATE OR REPLACE or not?

File: ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java (hunk 2 above)
Comment (agavra) [reply ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java:87]: they all actually return numbers, so it's kind-of irrelevant. I'll add it back for consistency, but we should probably change the API to just return a number.

File: ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java (hunk 3 above)
Comment (agavra) [reply ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java:64]: it'll actually fail before this point if it isn't a create or replace, but i'll make it explicit

Comment (agavra) [reply ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java:64]: On second thought, I'm going to leave it like that - whether or not the statement is `CREATE OR REPLACE` is not the responsibility of the queryID generation. If it's attempting to create an already existing statement without `OR REPLACE` it should fail in the metastore (`putSource` specifically)

File: ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java (hunk 2 above)
Comment (vcrfxia) [reply ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java:87]: Looks like it was added back in one place but not another (https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java#L58) :) If they're always numbers I'm fine with leaving it off, though.

Comment (agavra) [reply ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java:87]: 🤦 shouldn't make a difference, but i'll change it in the next PR to be safe

**Pull request merged**